libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/filescan/FileSystemScanExecutor.java

```diff
@@ -136,6 +142,111 @@ public void execute() {
         }
 
         mergeDuplicates(inventory);
+
+        // post-processing steps
+        // 1. produce one file with all ArtifactFile types
+        final File baseDir = fileSystemScanContext.getBaseDir().getFile();
+
+        List<FilePatternQualifierMapper> filePatternQualifierMappers = ComponentPatternValidator.detectDuplicateComponentPatternMatches(fileSystemScanContext.getScanParam().getReferenceInventory(), inventory, baseDir);
+
+
+        // 2. analyze component containments
+        for (FilePatternQualifierMapper mapper : filePatternQualifierMappers) {
+            final String assetId = "AID-" + mapper.getArtifact().getId() + "-" + mapper.getArtifact().getChecksum();
+            if (mapper.getSubSetMap() != null) {
+                for (String qualifier : mapper.getSubSetMap().keySet()) {
+                    Artifact foundArtifact = fileSystemScanContext.getInventory().getArtifacts().stream().filter(a -> a.getId().equals(qualifier) || qualifier.equals(a.getComponent() + "-" + a.getId() + "-" + a.getVersion())).findFirst().orElse(null);
```

Comment:
improve readability; we have variables with expressive names

Reply:
are you refering to line 158 and its length? or also the variable name?